### ISSUE TITLE
Conflict with Gravity Forms file upload field

### ISSUE THREAD

**Issue by robincornett** [ultimatemember/ultimatemember#157]
I have a site running Gravity Forms and Ultimate Member. On a page with a Gravity Form with a File Upload field, an error is consistently returned in the browser when the upload is attempted:

`Error: -200, Message: HTTP error, File: file.pdf`

There is also a 500 error in the console log. I submitted a ticket to Gravity, and they confirmed the issue, and added this: "The response also includes the text "Hello, spam bot!"." ... I searched through the code and found this message in `ultimate-member/core/um-form.php`, in a few places, so I started looking into that--but it seems like the short version is that UM is running security checks on a form it doesn't own?

The first error in my server log reads:

```
PHP Notice:  Undefined index: mode in ...plugins/ultimate-member/core/um-form.php on line 124
```

(reference: [line 124](https://github.com/ultimatemember/ultimatemember/blob/master/core/um-form.php#L124)) There is a similar message for `timestamp` on line 126.

(Note: on the page in question, there are no UM forms, not login or anything.)

If I add in an early return before line 124 based on `isset( $this->form_data['mode'] )`, the Gravity file upload field works as expected. However, since the method is all about security and preventing bad things from happening, I figured it was better to report this here.

UM version is 1.3.66; Gravity version is 2.0.3.

**Comment by champsupertramp** [comment]
Hi,

Have you tried adding a hidden field with name 'mode' and value 'true' ?

Regards,

**Comment by robincornett** [comment]
I just tried that here and the error messages are unchanged. Still undefined index for mode on line 124 and timestamp on 126.

**Comment by champsupertramp** [comment]
Hi,

Please check the latest version 1.3.72 and see if this resolves the issue.

Regards,

**Comment by robincornett** [comment]
The pre-release version I downloaded does not fix the issue, unfortunately. I did some more digging and only the multiple file upload field is affected, probably because that uses ajax? Changing the field to a single file upload bypasses the issue, since it's not uploaded until the form is submitted, is my guess. I'm recommending to my client that they use that upload option instead for the time being.

Note: the errors are unchanged except for the line numbers. Undefined index: mode is now line 145; timestamp is line 147.
